### ISSUE TITLE
Signing in not working?

### ISSUE THREAD

**Issue by AStartledRaccoon** [ShinjiruApp/Shinjiru#31]
When I open Shinjiru and try to sign in, my username and password is never recognised, I have tried with using my email and with my username. I have reset my password and tried with multiple browsers. I am using Ubuntu 18.04, and running Shinjiru v3.3.3. When I cancel the error message reads {"error":"invalid_client","message":"Client authentication failed"}. Thanks for reading!

**Comment by Kazakuri** [comment]
Hey there, sorry about that! 

If it's showing up with the pop-up login box then something is definitely going wrong. I'll look into it more when I get home in a few hours, but can you check the URL it's opening and make sure that it's:

```
https://anilist.co/api/v2/oauth/authorize?client_id=16&redirect_uri=http://localhost:42069/auth&response_type=code
```

The AniList API seems to behave a bit weirdly if the parameters aren't quite right. What I suspect might be happening is that the `redirect_uri` is getting set to `127.0.0.1` for you instead of `localhost`.

**Comment by AStartledRaccoon** [comment]
Thank you for responding so fast! 
As you thought, it does redirect seem to get set to 127.0.0.1. The url is https://anilist.co/api/v2/oauth/authorize?client_id=16&redirect_uri=http://localhost:42069/auth&response_type=code, and ![this](https://user-images.githubusercontent.com/43450024/52742389-8b1a4080-2fcf-11e9-9513-f2a4c0c4033e.png) is what pops up.

 If I manually go to the URL you linked it sends me to the authorisation page and says I have recieved the callback, but Shinjiru still doesn't recognise me.

 I also tried disabling my vpn but that didn't appear to have an effect. I'll try some other things and let you know if they work. Thanks for the help though!

**Comment by Kazakuri** [comment]
Hmmn, what are the contents of the log file? `~/.Shinjiru.log`

**Comment by AStartledRaccoon** [comment]
I can't seem to find the Shinjiru.log file. It's not in my home directory like its supposed to be and after searching for it I haven't been able to see anything either.

**Comment by Kazakuri** [comment]
Just to make sure, the file does start with a dot so it might not be visible by default in most file explorers.

I'll setup an 18.04 VM tonight to look into it more.

**Comment by Kazakuri** [comment]
It appears that the Qt Network Authorization library [committed a breaking change](https://github.com/qt/qtnetworkauth/commit/c85304484261af2fc046c909087549799a5979f9) that was released with Qt v5.12.1 and causes `127.0.0.1` to be used instead of `localhost` for the `redirect_uri`.

For now downgrading Qt to 5.12.0 through the Qt Maintenance Tool will let it run and authenticate properly.

It's not really feasible for me to change the client's `redirect_uri` since existing versions of Shinjiru on other platforms rely on `localhost` instead of `127.0.0.1`.
As a more permanent fix I'll add a preprocessor check that migrates to a new AniList client for newer Qt versions over the weekend.

**Comment by AStartledRaccoon** [comment]
Sorry for the late response, but your solution worked perfectly. Thank you very much! Also, I did manage to find the log file, it was hidden like you suspected. 
Here are the contents:
`[2019-02-14 13:36:38.945] Debug: Unable to open file: "/home/callum/.config/Kazakuri/Shinjiru/themes/Default.qss"
[2019-02-14 13:36:39.293] Debug: File not found "/home/callum/.local/bin/customMediaPlayers.json"
[2019-02-14 13:36:39.302] Debug: QNetworkReplyHttpImplPrivate::_q_startOperation was called more than once QUrl("https://shinjiru.me/static/changes.json")`

**Comment by Kazakuri** [comment]
Glad to hear it's working 👍 

Pushed c69bcb7 that fixes authentication for Qt 5.12.1 and later too.
